.github/workflows/main.yml

```diff
@@ -59,3 +59,41 @@ jobs:
       - name: Run pytest
         run: |
           pytest -v test/
+
+  lint:
+    runs-on: ubuntu-22.04
+    name: Lint
+
+    steps:
+      - name: Check out source code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # This is necessary to get the commits
```

Comment:
The issue with `0` is you get the full history, which can take a lot of time and isn't useful past the commits in the PR.

Reply:
I need to fix this.